core/src/main/java/com/gentics/mesh/dagger/module/OverridesProviderModule.java

```diff
@@ -0,0 +1,48 @@
+package com.gentics.mesh.dagger.module;
+
+import java.util.function.Function;
+
+import javax.annotation.Nullable;
+import javax.inject.Singleton;
+
+import com.gentics.mesh.cli.BootstrapInitializer;
+import com.gentics.mesh.core.search.index.node.NodeIndexHandler;
+import com.gentics.mesh.dagger.CoreMeshComponent;
+import com.gentics.mesh.dagger.OverrideSupplier;
+import com.gentics.mesh.search.index.node.NodeIndexHandlerImpl;
+import com.gentics.mesh.util.SearchWaitUtil;
+import com.gentics.mesh.util.SearchWaitUtilImpl;
+
+import dagger.Module;
+import dagger.Provides;
+
+@Module
+public class OverridesProviderModule {

```

Comment:
The base of the dep override mechanism. Since Mesh provides all the required deps with getters, we can use it to initialize the overridable/overridden deps manually.